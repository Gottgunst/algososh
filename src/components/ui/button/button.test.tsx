import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Button } from './button';

// Тестирование компонента Button
describe('Button', () => {
  // Проверяем при помощи снэпшотов корректную отрисовку:
  // кнопки с текстом;
  it('Render with Text', () => {
    //############################## arrange
    const button = renderer.create(<Button text='Развернуть' />).toJSON();
    //############################## assert
    expect(button).toMatchSnapshot();
  });

  // кнопки без текста;
  it('Render without Text', () => {
    //############################## arrange
    const button = renderer.create(<Button />).toJSON();
    //############################## assert
    expect(button).toMatchSnapshot();
  });

  // заблокированной кнопки;
  it('Render disabled', () => {
    //############################## arrange
    const button = renderer
      .create(<Button text='Развернуть' disabled />)
      .toJSON();
    //############################## assert
    expect(button).toMatchSnapshot();
  });

  // кнопки с индикацией загрузки.
  it('Render isLoading', () => {
    //############################## arrange
    const button = renderer
      .create(<Button text='Развернуть' isLoader={true} />)
      .toJSON();
    //############################## assert
    expect(button).toMatchSnapshot();
  });

  // Проверяем корректность вызова колбека при клике на кнопку.
  it('Correct run callback onClick', () => {
    //############################## arrange
    const clearItem = jest.fn();
    render(<Button text='Очистить' onClick={clearItem} />);
    const button = screen.getByText('Очистить');

    //############################## act
    fireEvent.click(button);

    //############################## assert
    expect(clearItem).toHaveBeenCalled();
  });
});
