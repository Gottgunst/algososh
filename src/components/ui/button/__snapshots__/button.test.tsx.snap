// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Render disabled 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  >
    Развернуть
  </p>
</button>
`;

exports[`Button Render isLoading 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Button Render with Text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Развернуть
  </p>
</button>
`;

exports[`Button Render without Text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;
